methods.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  def my_each
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    for j in self
+      yield j, i
+      i += 1
+    end
+  end
+
+  def my_select
+    result = []
+    self.my_each do |i|
+      result.<<(i) if yield(i)
+    end
+    result
+  end
+
+  def my_all?
+    self.my_each { |i| return true if yield(i) }
+    false
+  end
+
+  def my_any?
+    self.my_each { |i| return true if yield(i) }
+    false
+  end
+
+  def my_none?
+    self.my_each { |i| return false if yield(i) }
+    true
+  end
+
+  def my_count(item = nil)
+    count = 0
+    for i in self
+      if !item.nil?
+        count += 1 if i == item
+      elsif block_given?
+        count += 1 if yield i
+      else
+        count += 1
+      end
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    arry = []
+    if !proc.nil?
+      for i in self
+        element = proc.call(i)
+        arry << element
+      end
+    else
+      for i in self
+        element = yield i
+        arry << element
+      end
+    end
+    arry
+  end
+
+  def my_inject(initial = 0)
```

Comment:
this method also throws this errror    
```NoMethodError:
        undefined method `length' for 5..50:Range
```

Reply:
Tested with `(5..10).inject(1) { |product, n| product * n }` & `(5..10).inject { |sum, n| sum + n } ` and works now